MatrixSDK/Crypto/V2/MXCryptoMachine.swift

```diff
@@ -0,0 +1,206 @@
+//
+//  CryptoBridge.swift
+//  MatrixSDK
+//
+//  Created by Element on 05/05/2022.
+//
+
+import Foundation
+
+#if DEBUG && os(iOS)
+
+import MatrixSDKCrypto
+
+/// Wrapper around Rust-based `OlmMachine`, providing a more convenient API.
+///
+/// Two main responsibilities of the `MXCryptoMachine` are:
+/// - mapping to and from raw strings passed into the Rust machine
+/// - performing network requests and marking them as completed on behalf of the Rust machine
+@available(iOS 13.0.0, *)
+class MXCryptoMachine {
+    private static let storeFolder = "MXCryptoStore"
+    
+    enum Error: Swift.Error {
+        case invalidStorage
+        case invalidEvent
+        case nothingToEncrypt
+    }
+    
+    var deviceCurve25519Key: String? {
+        guard let key = machine.identityKeys()["curve25519"] else {
+            log(error: "Cannot get device curve25519 key")
+            return nil
+        }
+        return key
+    }
+    
+    var deviceEd25519Key: String? {
+        guard let key = machine.identityKeys()["ed25519"] else {
+            log(error: "Cannot get device ed25519 key")
+            return nil
+        }
+        return key
+    }
+    
+    private let machine: OlmMachine
+    private let requests: MXCryptoRequests
+
+    init(userId: String, deviceId: String, restClient: MXRestClient) throws {
+        requests = MXCryptoRequests(restClient: restClient)
+    
+        let url = try Self.storeURL(for: userId)
+        machine = try OlmMachine(
+            userId: userId,
+            deviceId: deviceId,
+            path: url.path,
+            passphrase: nil
+        )
+        
+        setLogger(logger: self)
+    }
+    
+    static func storeURL(for userId: String) throws -> URL {
+        guard let sharedContainerURL = FileManager.default.applicationGroupContainerURL() else {
```

Comment:
It makes sense to provide another directory for storage even if `MXSDKOptions.sharedInstance().applicationGroupIdentifier` is not set. I believe we won't need `invalidStorage` error in this case.

Reply:
Good point, the existing `MXFileStore` and `RealmCryptoStore` use different alternative folders, so here I am just using the user's documents directory as an alternative